doc/mgmt/gnmi/gnoi_os_reboot_file_reset.md

```diff
@@ -0,0 +1,449 @@
+  // When booting the requested OS version fails, the Target recovers by
+  // booting the previously running OS package.
+  rpc Activate(ActivateRequest) returns (ActivateResponse);
+
+  // Verify checks the running OS version. During reboot, gRPC client returns
+  // the gRPC status code UNAVAILABLE while the Target is unreachable, which
+  // should be retried by the client until successful. After the Target becomes
+  // reachable, it should report all ready or error states normally through
+  // VerifyResponse.
+  //
+  // On a dual Supervisor system, if the Standby Supervisor is rebooting, it
+  // should be reported in the VerifyResponse via verify_standby as
+  // StandbyState UNAVAILABLE.
+  rpc Verify(VerifyRequest) returns (VerifyResponse);
+}
+```
+
+#### gNOI OS Sequence Flow
+
+![gnoi_install](https://github.com/user-attachments/assets/6f3c40b1-c0bc-4ea9-a991-8d0deea05a53)
+
+Step 1. OS Install on the primary and secondary controller cards.
+
+Install and activate a new software version on primary supervisor.
+
+a. Issue gnoi.os.Install RPC to the chassis with InstallRequest.TransferRequest message. The message should set the version to the desired new version image, and standby_supervisor to FALSE.
+
+Wait for the switch to respond with InstallResponse. Expect it to return TransferReady.
+
+b. Transfer the content by issuing gnoi.os.Install rpc with InstallRequest.transfer_content message. 
+  * Expect it to return InstallResponse with a TransferProgress status asynchronously at certain intervals.
+
+c. End the transfer of software by issuing gnoi.os.Install rpc with InstallRequest.TransferEnd message. 
+  * Expect the switch to return InstallResponse with a Validated message. The version in the message should be set to the one which was transferred above.
+
+d. Activate the software by issuing gnoi.os.Activate rpc. 
+  * Set the version field of the ActivateRequest message to be the same as the version specified in the TransferRequest message above.
+  * Set the no_reboot flag to true.
+  * Set the standby_supervisor to FALSE.
+
+2. Install and activate the same new software version on standby supervisor:
+
+a. Repeat the above process of TransferRequest. This time set the standby_supervisor to TRUE.
+
+* Expect the switch to return a InstallResponse with a SyncProgress message. The switch should sync the software image from primary SUP to standby.
+* Expect the sync to return a value of 100 for percentage_transferred field.
+* At the end, expect the switch to return InstallResponse with a Validated message. The version in the message should be set to the one which was transferred above.
+
+3. Activate the software by issuing gnoi.os.Activate rpc as in the case of primary supervisor.
+
+a. Set the version field of the ActivateRequest message to be the same as the version specified in the TransferRequest message above.
+
+b. Set the no_reboot flag to true.
```

Comment:
Is this a server-side requirement (so as the standby_supervisor)? I am all for this but we should probably document, so client don't make unsupported rpc calls.